src/prefect/tasks/jupyter/jupyter.py

```diff
@@ -53,17 +59,20 @@ def run(
         - path (string, optional): path to fetch the notebook from; can also be
             a cloud storage path
         - parameters (dict, optional): dictionary of parameters to use for the notebook
-        - output_format (str, optional): Notebook output format.
-            Currently supported: json, html (default: json)
+        - output_format (str, optional): Notebook output format, should be a valid
+            nbconvert Exporter name.
+            Valid exporter names: asciidoc, custom, html, latex, markdown,
+            notebook, pdf, python, rst, script, slides, webpdf. (default: notebook)
+        - exporter_kwargs (dict, optional): The arguments used for initializing 
+            the exporter.
         """
         nb: nbformat.NotebookNode = pm.execute_notebook(
             path, "-", parameters=parameters, kernel_name=self.kernel_name
         )
-        if output_format == "json":
-            return nbformat.writes(nb)
-        if output_format == "html":
-            html_exporter = nbconvert.HTMLExporter()
-            (body, resources) = html_exporter.from_notebook_node(nb)
-            return body
 
-        raise NotImplementedError("Notebook output %s not supported", output_format)
+        if exporter_kwargs is None:
+            exporter_kwargs = {}
+
+        exporter = nbconvert.get_exporter(output_format)
```

Comment:
For backwards compatibility we'll still need to handle `"json"` as an alias of `notebook`. No need to document it in the docstring, just something like:

```
if output_format == "json":
    output_format = "notebook"
exporter = nbconvert.get_exporter(output_format)
```